packages/frontend-2/components/projects/MoveToWorkspaceDialog.vue

```diff
@@ -131,48 +121,49 @@ const versionsText = computed(() =>
   props.project.versions.totalCount === 1 ? 'version' : 'versions'
 )
 const dialogButtons = computed<LayoutDialogButton[]>(() => {
-  if (isCreatingWorkspace.value) return []
-
-  return [
-    {
-      text: 'Cancel',
-      props: { color: 'outline' },
-      onClick: () => {
-        open.value = false
-      }
-    },
-    {
-      text: 'Move',
-      props: {
-        color: 'primary',
-        disabled: !selectedWorkspace.value && !props.workspace
-      },
-      onClick: () => {
-        if (props.project && (selectedWorkspace.value || props.workspace)) {
-          const workspaceId = (selectedWorkspace.value?.id ||
-            props.workspace?.id) as string
-          const workspaceName =
-            selectedWorkspace.value?.name || (props.workspace?.name as string)
-
-          moveProject({
-            projectId: props.project.id,
-            workspaceId,
-            workspaceName,
-            eventSource: props.eventSource
-          })
-
-          open.value = false
+  return hasWorkspaces.value
+    ? [
+        {
+          text: 'Cancel',
+          props: { color: 'outline' },
+          onClick: () => {
+            open.value = false
+          }
+        },
+        {
+          text: 'Move',
+          props: {
+            color: 'primary',
+            disabled: !selectedWorkspace.value && !props.workspace
+          },
+          onClick: () => onMoveProject()
         }
-      }
-    }
-  ]
+      ]
+    : [
+        {
+          text: 'Close',
+          props: { color: 'outline' },
+          onClick: () => {
+            open.value = false
+          }
+        }
+      ]
 })
 
-const onWorkspaceCreated = (
-  workspace: ProjectsMoveToWorkspaceDialog_WorkspaceFragment
-) => {
-  isCreatingWorkspace.value = false
-  selectedWorkspace.value = workspace
+const onMoveProject = () => {
+  const workspaceId = selectedWorkspace.value?.id ?? props.workspace?.id
+  const workspaceName = selectedWorkspace.value?.name ?? props.workspace?.name
+
+  if (workspaceId && workspaceName) {
+    moveProject({
```

Comment:
shouldnt this be awaited before closing the dialog?

Reply:
yeah good point, updated